src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/FileValidator.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validates that a recipe or deployment bundle option with a FilePath typehint points to an actual file.
+    /// </summary>
+    public class FileValidator : IOptionSettingItemValidator
+    {
+        public static readonly string defaultValidationFailedMessage = "The specified file does not exist";
+
+        public string ValidationFailedMessage { get; set; } = defaultValidationFailedMessage;
+
+        ValidationResult IOptionSettingItemValidator.Validate(object input)
+        {
+            IFileManager fileManager = new FileManager();
+            var userFilePath = input?.ToString() ?? string.Empty;
+
+            // Allow the user to clear the option intentionally
+            if (userFilePath == string.Empty)
+            {
+                return ValidationResult.Valid();
+            }
+
+            // Otherwise if there is a value, verify that it points to an actual file
+            if (fileManager.Exists(userFilePath))
```

Comment:
If `userFilePath` is relative the absolute path should be computed based on the current project directory set by either `--project-path` switch or during server mode. This will fail in this case because it will compute based on the current working directory which is not necessary the project directory.